app/src/main/java/ch/sdp/vibester/activity/profile/ProfileActivity.kt

```diff
@@ -1,204 +1,226 @@
+import android.view.View.*
+import android.view.Window
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.NestedScrollView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.sdp.vibester.R
+import ch.sdp.vibester.activity.MainActivity
+import ch.sdp.vibester.auth.FireBaseAuthenticator
+import ch.sdp.vibester.database.DataGetter
+import ch.sdp.vibester.database.ImageGetter
+import ch.sdp.vibester.helper.ImageHelper
+import ch.sdp.vibester.helper.IntentSwitcher
+import ch.sdp.vibester.user.OnItemClickListener
+import ch.sdp.vibester.user.ProfileFollowingAdapter
+import ch.sdp.vibester.user.User
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.qrcode.QRCodeWriter
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import dagger.hilt.android.AndroidEntryPoint
+import net.glxn.qrgen.android.MatrixToImageWriter
+import javax.inject.Inject
+
+/**
+ * Display user profile's data (image, username, scores, etc.) in UI
+ */
+@AndroidEntryPoint
+open class ProfileActivity : AppCompatActivity(), OnItemClickListener {
+    @Inject
+    lateinit var dataGetter: DataGetter
+
+    @Inject
+    lateinit var authenticator: FireBaseAuthenticator
+
+    @Inject
+    lateinit var imageGetter: ImageGetter
+
+    private val imageSize = 1000
+    val imageRequestCode = 100
+
+    private var followings: MutableList<User> ? = null
+    private var profileFriendsAdapter: ProfileFollowingAdapter?= null
+
+    /**
+     * Generic onCreate method belonging to ProfileActivity.
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        super.onCreate(savedInstanceState)
+        supportActionBar?.hide()
+        setContentView(R.layout.activity_profile)
+
+        setupRecycleViewForFriends()
+        followings = ArrayList()
+
+        setRetToMainBtnListener()
+        setFollowingScoresBtnListener(R.id.profile_scores, R.id.profile_scroll_stat, R.id.profile_scroll_following)
+        setFollowingScoresBtnListener(R.id.profile_following, R.id.profile_scroll_following, R.id.profile_scroll_stat)
+
+        queryDatabase()
+    }
+
+    private fun setupRecycleViewForFriends() {
+        findViewById<RecyclerView>(R.id.profile_followingList).apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = followings?.let { ProfileFollowingAdapter(it, this@ProfileActivity) }
+            setHasFixedSize(true)
+        }
+    }
+
+    private fun showFriendsPosition(friends: MutableList<User>?) {
+        profileFriendsAdapter = ProfileFollowingAdapter(friends!!, this)
+        findViewById<RecyclerView>(R.id.profile_followingList)!!.adapter = profileFriendsAdapter
+    }
+
+
+    //check the UID here not sure
```

Comment:
What is the purpose of this comment? is doubt resolve? Should we keep it ? 
If this is the case maybe add a // TODO

Reply:
I don't know... I didn't write this part.